src/plugins/files/server/file_client/utils.ts

```diff
@@ -19,3 +21,31 @@ export function createDefaultFileAttributes(): Pick<
     Updated: dateString,
   };
 }
+
+export const fetchDoc = async <TDocument = unknown>(
+  esClient: ElasticsearchClient,
+  index: string,
+  docId: string,
+  indexIsAlias: boolean = false
+): Promise<GetResponse<TDocument> | undefined> => {
+  if (indexIsAlias) {
+    const fileDocSearchResult = await esClient.search<TDocument>({
+      index,
+      body: {
+        size: 1,
+        query: {
+          term: {
+            _id: docId,
+          },
+        },
+      },
+    });
+
+    return fileDocSearchResult.hits.hits[0] as GetResponse<TDocument>;
+  }
+
+  return esClient.get<TDocument>({
```

Comment:
With this change, we do not have the same error behaviour non-existent docs. A call to `.get` will throw where a call to `.search` will not.

I think we should just copy what `x-pack/plugins/security_solution/server/endpoint/services/actions/action_files.ts` is doing in this case.